x-pack/plugins/logstash/server/plugin.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import {
+  CoreSetup,
+  ICustomClusterClient,
+  Logger,
+  Plugin,
+  PluginInitializerContext,
+} from 'src/core/server';
+import { LicensingPluginSetup } from '../../licensing/server';
+import { SecurityPluginSetup } from '../../security/server';
+
+import { registerRoutes } from './routes';
+
+interface SetupDeps {
+  licensing: LicensingPluginSetup;
+  security?: SecurityPluginSetup;
+}
+
+export class LogstashPlugin implements Plugin {
+  private readonly logger: Logger;
+  private esClient?: ICustomClusterClient;
+  constructor(context: PluginInitializerContext) {
+    this.logger = context.logger.get();
+  }
+
+  setup(core: CoreSetup, deps: SetupDeps) {
+    this.logger.debug('Setting up Logstash plugin');
+    const esClient = core.elasticsearch.createClient('logstash');
+
+    registerRoutes(core.http.createRouter(), esClient, deps.security);
```

Comment:
I know this forces to use a `setup/start` ping pong trick, but would it be possible to create the es client from the `start` contract instead to avoid using deprecated `setup` APIs?

Reply:
ok, makes sense